scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,365 @@
+    used in practice. Two more common methods are available, ``'basic'``
+    ('reverse percentile') and ``'BCa'`` ('bias-corrected and accelerated');
+    they differ in how step 3 is performed.
+
+    If the samples in `data` are  taken at random from their respective
+    distributions :math:`n` times, the confidence interval returned by
+    `bootstrap_ci` will contain the true value of the statistic for those
+    distributions approximately `confidence_level`:math:`\, \times \, n` times.
+
+    Parameters
+    ----------
+    data : sequence of array-like
+         Each element of data is a sample from an underlying distribution.
+    statistic : callable
+        Statistic for which the confidence interval is to be calculated.
+        `statistic` must be a callable that accepts ``len(data)`` samples
+        as separate arguments and returns the resulting statistic.
+        `statistic` must also accept a keyword argument `axis` and be
+        vectorized to compute the statistic along the provided `axis`.
+    axis : int, optional
+        The axis of the samples in `data` along which the `statistic` is
+        calculated. The default is ``0``.
+    confidence_level : float, optional
+        The confidence level of the confidence interval.
+        The default is ``0.95``.
+    n_resamples : int, optional
+        The number of resamples performed to form the bootstrap distribution
+        of the statistic. The default is ``1000``.
+    method : str in {'percentile', 'basic', 'bca'}, optional
+        Whether to return the 'percentile' bootstrap confidence interval
+        (``'percentile'``), the 'reverse'or the bias-corrected and accelerated
+        bootstrap confidence interval (``'BCa'``). The default is ``'BCa'``.
+        Note that only ``'percentile'`` and ``'basic'`` support multi-sample
+        statistics at this time.
+    random_state: int, RandomState, or Generator, optional
+        Pseudorandom number generator state used to generate resamples.
+
+    Returns
+    -------
+    ci_l : float
+        The lower bound of the confidence interval.
+    ci_u : float
+        The upper bound of the confidence interval.
+
+    References
+    ----------
+    .. [1] B. Efron and R. J. Tibshirani, An Introduction to the Bootstrap,
+       Chapman & Hall/CRC, Boca Raton, FL, USA (1993)
+    .. [2] Nathaniel E. Helwig, "Bootstrap Confidence Intervals",
+       http://users.stat.umn.edu/~helwig/notes/bootci-Notes.pdf
+    .. [3] Bootstrapping (statistics), Wikipedia,
+       https://en.wikipedia.org/wiki/Bootstrapping_(statistics)
+
+    Examples
+    --------
+    Suppose we have sampled data from an unknown distribution.
+
+    >>> import numpy as np
+    >>> np.random.seed(0)
+    >>> from scipy.stats import norm
+    >>> dist = norm(loc=2, scale = 4)  # our "unknown" distribution
+    >>> data = dist.rvs(size=100)      # a random sample from the distribution
```

Comment:
```suggestion
    >>> rng = np.random.default_rng()
    >>> from scipy.stats import norm
    >>> dist = norm(loc=2, scale = 4)  # our "unknown" distribution
    >>> # a random sample from the distribution
    >>> data = dist.rvs(size=100, random_state=rng)
```

Reply:
There are about a dozen other places where this is "needed" , so I'm going to take care of these locally.

Reply:
Ooops sorry if I missed some!